src/sentry/incidents/tasks.py

```diff
@@ -181,16 +182,24 @@ def handle_snuba_query_update(subscription_update, subscription):
     default_retry_delay=60,
     max_retries=5,
 )
-def handle_trigger_action(action_id, incident_id, method):
+def handle_trigger_action(action_id, incident_id, project_id, method):
     try:
-        action = AlertRuleTriggerAction.objects.get(id=action_id)
+        action = AlertRuleTriggerAction.objects.select_related(
+            "alert_rule_trigger", "alert_rule_trigger__alert_rule"
+        ).get(id=action_id)
     except AlertRuleTriggerAction.DoesNotExist:
         metrics.incr("incidents.alert_rules.skipping_missing_action")
         return
     try:
-        incident = Incident.objects.get(id=incident_id)
+        incident = Incident.objects.select_related("organization").get(id=incident_id)
     except Incident.DoesNotExist:
         metrics.incr("incidents.alert_rules.skipping_missing_incident")
         return
 
-    getattr(action, method)(incident)
+    try:
+        project = Project.objects.get(id=project_id)
```

Comment:
Should you use `get_from_cache` here? Is this attached to event post process at all?

Reply:
No, this is just attached to regular updates from query subscriptions. So every minute/few minutes an update will come through that may trigger this. We probably do need to work on scalability of this at some point though.